src/Amqp/JobQueue/AmqpWorker.php

```diff
@@ -0,0 +1,261 @@
+     * @param string $prefix A string to prefix to all method names.
+     */
+    public function exposeObject($object, $prefix = '')
+    {
+        $reflector = new ReflectionClass($object);
+
+        foreach ($reflector->getMethods() as $method) {
+            if ($method->isStatic()) {
+                continue;
+            } elseif (!$method->isPublic()) {
+                continue;
+            } elseif ('_' === $method->getName()[0]) {
+                continue;
+            }
+
+            $name = $prefix . $method->getName();
+
+            $this->expose(
+                $name,
+                [$object, $method->getName()]
+            );
+        }
+    }
+
+    /**
+     * Run the worker.
+     */
+    public function run()
+    {
+        $this->isStopping = false;
+
+        // Bind queues / consumers ...
+        foreach ($this->jobs as $jobName => $job) {
+            $this->bind($jobName);
+
+            $this->logger->debug(
+                'jobqueue.worker exposed job "{job}"',
+                ['job' => $jobName]
+            );
+        }
+
+        if ($this->jobs) {
+            $this->logger->info('jobqueue.worker started successfully');
+        } else {
+            $this->logger->warning('jobqueue.worker started without exposed jobs');
+        }
+
+        while ($this->channel->callbacks) {
+            $this->channelDispatcher->wait($this->channel);
+
+            if ($this->isStopping) {
+                foreach ($this->jobs as $jobName => $job) {
+                    $this->unbind($jobName);
+                }
+            }
+        }
+
+        $this->logger->info('jobqueue.worker shutdown gracefully');
+    }
+
+    /**
+     * Stop the worker.
+     */
+    public function stop()
+    {
+        if (!$this->isStopping) {
+            $this->isStopping = true;
+
+            $this->logger->info('jobqueue.worker stopping');
+        }
+    }
+
+    /**
+     * Receive a task request message.
+     *
+     * @param AMQPMessage $message
+     */
+    private function recv(AMQPMessage $message)
+    {
+        $logLevel = LogLevel::DEBUG;
+        $logContext = [
+            'task' => '<unknown>',
+            'payload' => '<unknown>',
+        ];
+
+        $logMessage = null;
+        $logCode = 0;
+
+        try {
+            $task = $this->serialization->unserializeRequest($message->body);
+
+            $logContext['task'] = $task->name();
+            $logContext['payload'] = json_encode($task->payload());
+
+            call_user_func(
+                $this->jobs[$task->name()],
+                $task->payload()
+            );
+
+            $logMessage = 'jobqueue.worker completed task {task}({payload})';
+            $this->channel->basic_ack($message->get('delivery_tag'));
+        } catch (InvalidMessageException $e) {
+            $logLevel = LogLevel::WARNING;
+            $logMessage = $this->handleFailure($message, $e, $logContext);
+        } catch(ErrorException $e) {
```

Comment:
Asplode converts errors to ErrorExceptions so invalid/missing parameters are getting caught here however I don't want to discard ErrorExceptions because of other errors that can occur.